training/distributed_training/pytorch/data_parallel/deepspeed/llama2/smddp_deepspeed_example.ipynb

```diff
@@ -0,0 +1,289 @@
+    "\n",
+    "*Note 2: The SMDDP library for accelerated sharded data parallel training is compatible with deep learning containers from PyTorch 2.0 onwards.  Ensure you are using PyTorch >=2.0 for this example.*"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Files ##\n",
+    "\n",
+    "All training and helper scripts are stored in the `code/` folder:\n",
+    "* `dsconfig.json` - Deepspeed config file \n",
+    "* `requirements.txt` - Dependencies for this example that will be installed on container when training job is launched\n",
+    "* `train.py` - Entry point training script\n",
+    "* `utils.py` - Defines dummy dataset and constructs dataloaders for the training job"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setting up"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "If you are going to use Sagemaker in a local environment, you need access to an IAM Role with the required permissions for Sagemaker. You can find more about it [here](https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html).\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "os.environ[\"PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION\"] = \"python\"\n",
+    "\n",
+    "import sagemaker\n",
+    "import boto3\n",
+    "\n",
+    "sagemaker_session = sagemaker.Session()\n",
+    "\n",
+    "try:\n",
+    "    role = sagemaker.get_execution_role()\n",
+    "except ValueError:\n",
+    "    iam = boto3.client(\"iam\")\n",
+    "    role = iam.get_role(RoleName=\"sagemaker_execution_role\")[\"Role\"][\"Arn\"]\n",
+    "\n",
+    "print(f\"sagemaker role arn: {role}\")\n",
+    "print(f\"sagemaker bucket: {sagemaker_session.default_bucket()}\")\n",
+    "print(f\"sagemaker session region: {sagemaker_session.boto_region_name}\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that SageMaker by default uses the latest [AWS Deep Learning Container (DLC)](https://aws.amazon.com/machine-learning/containers/), but if you want to use your own DLC, you can set the `use_ecr_image` flag to `True` and set the `ecr_image` variable. Also note that if using FSx when launching the SageMaker notebook instance, you will need to use the same `subnet` and `security_group_config`.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "use_ecr_image=False\n",
+    "use_fsx=False\n",
+    "kwargs = {}\n",
+    "\n",
+    "if use_ecr_image:\n",
+    "    ecr_image = \"<ECR_IMAGE_URI>\"\n",
```

Comment:
I find existing notebook adding examples of those placeholder vars, maybe we can do that as well.
- It's easier for the reader to follow then, e.g. (though I think the value is not really a valid example)

```
image = "<ADD NAME OF REPO>"  # Example: bert-smdataparallel-sagemaker (Change to a valid example/ format instead)
```